### ISSUE TITLE
apex gradient clipping

### ISSUE THREAD

**Issue by ruotianluo** [PyTorchLightning/pytorch-lightning#2556]
#🐛 Bug

It seems when apex + gradient clipping is not handled explicitly.
I see note here https://nvidia.github.io/apex/advanced.html#gradient-clipping.
where amp.master_params needs to be used.

Is what's in this repo correct or not?

**Comment by Borda** [comment]
@ruotianluo mind send a PR? 🐰
